### ISSUE TITLE
The `rangeStart` option of `event.occurrences` does not consider time.

### ISSUE THREAD

**Issue by msimpson** [solspace/craft-calendar#145]
It appears that the `rangeStart` option of `event.occurrences` no longer considers time.

This was not the case in earlier versions, therefore this is likely a regression.

**Steps to reproduce**

1. Create a calendar event which repeats daily at a time which is in the future.

2. Attempt to retrieve today's occurrence via `event.occurrences( { rangeStart: now, limit: 1 } )`

3. Receive tomorrows occurrence, instead, even though the time of today's occurrence has yet to happen.

**Expected behavior**

Receive today's occurrence, as expected, if the time component of `rangeStart` was being considered.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Craft Pro 3.7.42
- Calendar Version: 3.3.11
- Calendar Edition: Pro
- Fresh Install or Upgrade: Upgrade

**Additional context**

If the `rangeStart` option is removed, today's occurrence is returned.

However, today's occurrence is returned even if the end time of that occurrence has passed.

The only way to deal with the current behavior seems to be the following Twig implementation:

```
{% if event.repeating %}

  {% set occurrence = event.occurrences( { limit: 1 } ) | first %}

  {% if occurrence.startDateLocalized < now %}
  
    {% set occurrence = event.occurrences( { rangeStart: now, limit: 1 } ) | first %}
  
  {% endif %}

{% else %}

  {% set occurrence = event %}

{% endif %}
```

Also, note that [the documentation on `occurrences` and its options](https://docs.solspace.com/craft/calendar/v3/template-objects/occurrences.html#properties) is woefully inadequate.

There's nothing available to discern whether this is expected behavior or not.

I must rely solely on my prior experience with earlier versions of this plugin to determine this is aberrant behavior.

**Comment by kjmartens** [comment]
I'm very sorry for the delay @msimpson. I will have a developer check into this and see if we can see what's up. 🙂

**Comment by kjmartens** [comment]
I believe this might have been resolved with another issue that was fixed in 3.3.14+. If this is still an issue for you, would you be able to try updating to that version, and then use the new on-demand option (`addFirstOccurrenceDate()`) to display the first occurrence date result in an occurrences list of events that use the "Select Dates" repeat rule. 🙂
